Implementing an interface with generics results in builder error

<!-- Please make sure to run `v up` before reporting any issues as it may have already been fixed.
     It's also advisable to update all relevant modules using `v outdated` and `v install` -->

<!-- You can use `v doctor` to fill up the next fields -->
**V version:** 0.2.2 8c95f07
**OS:** cc (Ubuntu 9.3.0-17ubuntu1~20.04) 9.3.0


**What did you do?**
```v
// compile with v -shared -cg .
module test

interface Depends {
	depends() []Depends
}

struct Signal<T> {
}

fn (x Signal<T>) depends() []Depends {
	return []
}

struct Add<T> {
	a Signal<T>
	b Signal<T>
}

fn (a Add<T>) depends() []Depends {
	return [a.a, a.b]
}
```
 
**What did you see instead?**
```
/tmp/v/test.10625239837982459845.tmp.so.c:1524: error: ',' expected (got "x")
builder error: 
==================
C error. This should never happen.

If you were not working with C interop, this is a compiler bug, please raise an issue on GitHub:

https://github.com/vlang/v/issues/new/choose

You can also use #help on Discord: https://discord.gg/vlang
```

--- a/vlib/v/ast/table.v
+++ b/vlib/v/ast/table.v
@@ -1121,6 +1121,12 @@ pub fn (mut t Table) does_type_implement_interface(typ Type, inter_typ Type) boo
 	if typ_sym.language != .v {
 		return false
 	}
+	// generic struct don't generate cast interface fn
+	if typ_sym.info is Struct {
+		if typ_sym.info.is_generic {
+			return false
+		}
+	}
 	mut inter_sym := t.get_type_symbol(inter_typ)
 	if typ_sym.kind == .interface_ && inter_sym.kind == .interface_ {
 		return false

--- a/vlib/v/tests/generics_interface_decl_test.v
+++ b/vlib/v/tests/generics_interface_decl_test.v
@@ -1,0 +1,23 @@
+interface Depends {
+	depends() []Depends
+}
+
+struct Signal<T> {
+}
+
+fn (x Signal<T>) depends() []Depends {
+	return []
+}
+
+struct Add<T> {
+	a Signal<T>
+	b Signal<T>
+}
+
+fn (a Add<T>) depends() []Depends {
+	return [a.a, a.b]
+}
+
+fn test_generics_interface_decl() {
+	assert true
+}

